tests/regression/lp-1803542/task.yaml

```diff
@@ -0,0 +1,53 @@
+summary: Regression test for https://bugs.launchpad.net/snapd/+bug/1803542
+environment:
+    VARIANT/absent: absent
+    VARIANT/present: present
+    VARIANT/shared: shared
+    VARIANT/private: private
+prepare: |
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB/snaps.sh"
+    install_local test-snapd-sh
+    #shellcheck source=tests/lib/dirs.sh
+    . "$TESTSLIB/dirs.sh"
+    # Ensure that every snap namespace is discarded.
+    for name in $(snap list | awk '!/Name/ {print $1}'); do
+        "$LIBEXECDIR"/snapd/snap-discard-ns "$name"
+    done
+    if findmnt --noheadings /run/snapd/ns >/dev/null; then
+        umount /run/snapd/ns
+    fi
+    ! findmnt --noheadings /run/snapd/ns
+debug: |
+    cat /proc/self/mountinfo
+execute: |
+    case "$VARIANT" in
+       absent)
+            test -d /run/snapd/ns && rmdir /run/snapd/ns
+            ;;
+       present)
+            mkdir -p /run/snapd/ns
+            ;;
+       shared)
+            mkdir -p /run/snapd/ns
+            mount --bind /run/snapd/ns /run/snapd/ns
+            mount --make-shared /run/snapd/ns
+            # verify mount tags to ensure the mount point has shared propagation (format shared:<nnnn>)
+            test "$(grep '/snapd/ns /run/snapd/ns' < /proc/self/mountinfo | awk '{print $7}' | cut -d : -f 1)" = shared
+            ;;
+        private)
+            mkdir -p /run/snapd/ns
+            mount --bind /run/snapd/ns /run/snapd/ns
+            mount --make-private /run/snapd/ns
```

Comment:
Should we do anything about these mount points in `restore:`?

Reply:
Not really, snap-confine restores the correct setup on each execution. We don't have to unmount them. The code here is what snap-confine does internally.